resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -154,10 +156,24 @@ private[spark] class KubernetesClusterSchedulerBackend(
       } else if (totalExpectedExecutors.get() <= runningExecutorPods.size) {
         logDebug("Maximum allowed executor limit reached. Not scaling up further.")
       } else {
+        val executorPodsWithIPs = EXECUTOR_PODS_BY_IPS_LOCK.synchronized {
+            executorPodsByIPs.values.toList
+          }
+        val nodeToLocalTaskCount = mutable.Map[String, Int]() ++
+          KubernetesClusterSchedulerBackend.this.synchronized {
+            hostToLocalTaskCount
+          }
+        for (pod <- executorPodsWithIPs) {
+          nodeToLocalTaskCount.remove(pod.getSpec.getNodeName).nonEmpty ||
```

Comment:
This removes cluster nodes that are running our executor pods already since we don't need to launch more executors on those nodes. Added a comment.